src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -3,115 +3,96 @@
-  const [selectedOption, setSelectedOption] = useState<EuiComboBoxOptionOption<string>>();
-  const { view, viewRegistry } = useView();
-  const views = viewRegistry.all();
-  const viewOptions = useMemo(
-    () =>
-      views.map(({ id, title }) => ({
-        value: id,
-        text: title,
-      })),
-    [views]
-  );
+  const [selectedSources, setSelectedSources] = useState<DataSourceOption[]>([]);
+  const [dataSourceOptionList, setDataSourceOptionList] = useState<DataSourceGroup[]>([]);
+  const [activeDataSources, setActiveDataSources] = useState<DataSourceType[]>([]);
 
   const {
     services: {
-      data: { indexPatterns },
+      data: { indexPatterns, dataSources },
       notifications: { toasts },
+      application,
     },
   } = useOpenSearchDashboards<DataExplorerServices>();
 
   useEffect(() => {
     let isMounted = true;
-    const fetchIndexPatterns = async () => {
-      await indexPatterns.ensureDefaultIndexPattern();
-      const cache = await indexPatterns.getCache();
-      const currentOptions = (cache || []).map((indexPattern) => ({
-        label: indexPattern.attributes.title,
-        value: indexPattern.id,
-      }));
-      if (isMounted) {
-        setOptions(currentOptions);
+    const subscription = dataSources.dataSourceService.dataSources$.subscribe(
+      (currentDataSources) => {
+        if (isMounted) {
+          setActiveDataSources(Object.values(currentDataSources));
+        }
       }
-    };
-    fetchIndexPatterns();
+    );
 
     return () => {
+      subscription.unsubscribe();
       isMounted = false;
     };
-  }, [indexPatterns]);
+  }, [indexPatterns, dataSources]);
+
+  const getMatchedOption = (dataSourceList: DataSourceGroup[], ipId: string) => {
+    for (const dsGroup of dataSourceList) {
+      return dsGroup.options.find((item) => {
+        return item.value === ipId;
+      });
+    }
+  };
 
-  // Set option to the current index pattern
   useEffect(() => {
     if (indexPatternId) {
-      const option = options.find((o) => o.value === indexPatternId);
-      setSelectedOption(option);
+      const option = getMatchedOption(dataSourceOptionList, indexPatternId);
+      setSelectedSources(option ? [option] : []);
+    }
+  }, [indexPatternId, activeDataSources, dataSourceOptionList]);
+
+  const handleSourceSelection = (selectedDataSources: DataSourceOption[]) => {
+    if (selectedDataSources.length === 0) {
+      setSelectedSources(selectedDataSources);
+      return;
     }
-  }, [indexPatternId, options]);
+    // Temporary redirection solution for 2.11, where clicking non-index-pattern datasource
+    // will redirect user to Observability event explorer
+    if (selectedDataSources[0].ds?.getType() !== 'DEFAULT_INDEX_PATTERNS') {
+      return application.navigateToUrl(
+        `../observability-logs#/explorer?datasourceName=${selectedDataSources[0].label}&datasourceType=${selectedDataSources[0].type}`
```

Comment:
Another type issue `Property 'type' does not exist on type 'DataSourceOption'.ts(2339)`